modules/apps/asset/asset-info-impl/src/main/java/com/liferay/asset/info/internal/item/provider/AssetEntryInfoItemFieldSetProviderImpl.java

```diff
@@ -49,6 +55,32 @@
 public class AssetEntryInfoItemFieldSetProviderImpl
 	implements AssetEntryInfoItemFieldSetProvider {
 
+	@Override
+	public InfoFieldSet getInfoFieldSet(AssetEntry assetEntry) {
+		InfoFieldSet infoFieldSet = getInfoFieldSet(AssetEntry.class.getName());
+
+		Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
+		Set<AssetVocabulary> assetVocabularies = _getAssetVocabularies(
+			assetEntry);
+
+		InfoFieldSet categorizationInfoFieldSet = new InfoFieldSet(
+			InfoLocalizedValue.localize(getClass(), "categorization"),
+			"categorization");
+
+		for (AssetVocabulary assetVocabulary : assetVocabularies) {
+			categorizationInfoFieldSet.add(
+				new InfoField(
+					TextInfoFieldType.INSTANCE,
+					InfoLocalizedValue.localize(
+						getClass(), assetVocabulary.getTitle(locale)),
```

Comment:
Instead of doing this, create an InfoLocalizedValue that has all the titles for the available locales. Here is an example of how to do this without having to iterate and copy all the values: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/info/localized/DDMStructureNameInfoLocalizedValue.java
In your case it would require creating an AssetVocabularyTitleInfoLocalizedValue that receives an assetVocabulary in its constructor.